src/marten/handlers/concerns/schema_callbacks.cr

```diff
@@ -0,0 +1,165 @@
+module Marten
+  module Handlers
+    # Provides the ability to define schema validation callbacks.
+    #
+    # This module provides the ability to define callbacks that are executed before, after, after success or/and after
+    # failed validation of a schema.
+    # They allow to intercept the incoming request and to return early HTTP responses. TwFouro hooks
+    # are enabled by the use of this module:
+    # - `before_schema_validation` callbacks are executed before the validation of the schema
+    # - `after_schema_validation` callbacks are executed right after the validation of the schema
+    # - `after_successful_schema_validation` callbacks are executed right after the validation but
+    #    only if the schema is valid
+    # - `after_failed_schema_validation` callbacks are executed right after the validation but
+    #    only if the schema is not valid
+    module SchemaCallbacks
```

Comment:
Can we define this module under a `src/marten/handlers/schema/` folder instead of defining it next to the shared concerns (asking because this concern is really specific to the schema handler)?